url882f91ec97_WenboCao_Microsoft_Drone_tgz-pJ8-org_ejml_alg_dense_decomposition_qr_StabilityQRDecompositionJ8/src/org/ejml/ops/TestConvertMatrixType.java

```diff
@@ -0,0 +1,192 @@
+/*
+ * Copyright (c) 2009-2013, Peter Abeles. All Rights Reserved.
+ *
+ * This file is part of Efficient Java Matrix Library (EJML).
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.ejml.ops;
+
+import org.ejml.alg.block.BlockMatrixOps;
+import org.ejml.data.BlockMatrix64F;
+import org.ejml.data.DenseMatrix64F;
+import org.ejml.data.FixedMatrix64F;
+import org.ejml.data.Matrix64F;
+import org.junit.Test;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.Random;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * @author Peter Abeles
+ */
+public class TestConvertMatrixType {
+
+    Random rand = new Random(234);
+
+    @Test
+    public void any_to_any() {
+        DenseMatrix64F a = new DenseMatrix64F(2,3,true,1,2,3,4,5,6);
+        DenseMatrix64F b = new DenseMatrix64F(2,3);
+
+        ConvertMatrixType.convert((Matrix64F)a,(Matrix64F)b);
+
+        assertTrue(MatrixFeatures.isIdentical(a,b,1e-12));
+    }
+
+    @Test
+    public void checkAll_Fixed_to_DM() throws IllegalAccessException, InstantiationException, InvocationTargetException {
+        Method[] methods = ConvertMatrixType.class.getMethods();
+
+        int numFound = 0;
+
+        for( Method m : methods ) {
+            if( !m.getName().equals("convert") )
+                continue;
+            Class[]param = m.getParameterTypes();
+
+            if( !FixedMatrix64F.class.isAssignableFrom(param[0]) ) {
+                continue;
+            }
+
+            FixedMatrix64F a = (FixedMatrix64F)param[0].newInstance();
+            DenseMatrix64F b = new DenseMatrix64F(a.getNumRows(),a.getNumCols());
+
+            for( int i = 0; i < b.numRows; i++ ) {
+                for( int j = 0; j < b.numCols; j++ ) {
+                    a.set(i, j, rand.nextDouble());
+                }
+            }
+
+            Object[] input = new Object[param.length];
+            input[0] = a;
+            input[1] = b;
+
+            m.invoke(null,input);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Catch `InvocationTargetException` explicitly when you call `Method.invoke()`. Use `Throwable.getCause()` or `getTargetException()` in the catch handler to access more information about the underlying exception. See the following diffs on GitHub that make this code change: [diff1](https://github.com/EsfingeFramework/gamification/commit/20792047e6c69457cca03e877f2ada578188f32a#diff-6808c12fe9ddaea042db84f95dd5ba78adab99d0cb18bb4d85916a8723530768L21), [diff2](https://github.com/cucumber-attic/cuke4duke/commit/f4821d400cf2590126f1298a8dff9f7a33d45dc5#diff-80adaffc20c0e6e7d2e42312386ffe1ddc10213bf5b8e8919895fd85d666c8deL34)

Learn more: [InvocationTargetException in the Java Platform, Standard Edition 8 API Specification](https://docs.oracle.com/javase/8/docs/api/java/lang/reflect/InvocationTargetException.html)